Elastic indices information (#480)
ElrondNetwork/docs.elrond.com: docs/sdk-and-tools/indices/blocks.md

---

> id: blocks

Author: done

> title: blocks

Author: done

---

> The `_id` field for this index is composed of block hash hex encoded

Reviewer: The `_id` field of this index is represented by the block hash, in a hexadecimal encoding

Author: done

| Field                 | Description                                                                                                |
|-----------------------|------------------------------------------------------------------------------------------------------------|

> | nonce                 | The nonce field represents sequence number of block.                                                       |

Reviewer: the sequence, the block

Author: done

> | round                 | The round field represents the round when block was proposed and executed.                                 |

Reviewer: the block

Author: done

> | epoch                 | The epoch field represent the epoch when the block was proposed and executed.                              |

Reviewer: represents

Author: done

> | miniBlocksHashes      | The miniBlocksHashes field represents the hashed hex encoded of the miniblocks that was included in block. |

Reviewer: The miniBlocksHashes field contains an array of the miniblock hashes (hexadecimal encoded) that were included in the block.

Author: done

> | miniBlocksDetails     | The miniBlocksDetails field represents the details of all the miniblocks that was included in block.       |

Reviewer: The miniBlocksDetails field contains an array of structures indicating processing details of the miniblocks, such as the index of the first processed transaction

> | notarizedBlocksHashes | The notarizedBlocksHashes field represents the hashes of the blocks the was notarized in current block.    |

Reviewer: that was*
the current block*

> | proposer              | The proposer field represents the index of the validator that proposed this block.                         |

Reviewer: this block ->  the block

Author: done

Reviewer: not done yet

Author: fixed

> | validators            | The validators field represents the indices on the validators that have signed for this block.             |

Reviewer: The validators field is an array that contains the indices of the validators that signed the block. The indices are to be found in the `validators` index.

Author: done.

> | notarizedBlocksHashes | The notarizedBlocksHashes field represents the hashes of the blocks that was notarized in the current block.                                                       |

Reviewer 2: ```suggestion
| notarizedBlocksHashes | The notarizedBlocksHashes field represents the hashes of the blocks that were notarized in the current block.                                                       |
```

Author: done

| pubKeyBitmap          | The pubKeyBitmap field represents the pub key bitmap.                                                      |

> | size                  | The size field represents the size of block in bytes.                                                      |

Reviewer: the block

Author: done

> | sizeTxs               | The sizeTxs field represents the size of transactions from block in bytes.                                 |

Reviewer: The sizeTxs field holds the size of the block's transactions in bytes

Author: done

> | timestamp             | The timestamp field represents the timestamp when block was proposed and executed.                         |

Reviewer: the block

Author: done

> | stateRootHash         | The stateRootHash field represents the state root hash of tree when block was proposed and executed.       |

Author: done

| prevHash              | The prevHash field represents the hash of the previous block.                                              |

> | shardId               | The shardId field represents the shard id of block.                                                        |

Reviewer: The shardId field represents the shard where the block was executed.

Author: done

> | txCount               | The txCount field represents the number of transactions that was executed in block.                        |

Reviewer: the block

> | txCount               | The txCount field represents the number of transactions that was executed in block.                        |

Author: done

> | notarizedTxsCount     | The notarizedTxsCount field represents the number of transactions that was notarized in block.             |

Reviewer: were

Reviewer: the block

> | accumulatedFees       | The accumulatedFees field represents the accumulated fee that was payed in block.                          |

Reviewer: the block

Author: done

> | developerFees         | The developerFees field represents the fee that was accumulated in block.                                  |

Author: done

| epochStartBlock       | The epochStartBlock is true if the current block is a start of epoch block.                                |

> | epochStartInfo        | The epochStartInfo field contains data about current epoch.                                                |

Author: done

> | gasProvided           | The gasProvided field represents the total gas that was provided in block.                                 |

Author: done

> | shardId               | The shardId field represents the shard where the block was executed.                                                                                               |

Reviewer 2: [completely optional]

```suggestion
| shardId               | The shardId field represents in which shard the block was executed.                                                                                               |
```

Author: done

> | gasRefunded           | The gasRefunded field represents the total gas that was refunded in block.                                 |

Author: done

> | gasPenalized          | The gasPenalized field represents the total gas that was penalized in block.                               |

Author: done

> | maxGasLimit           | The maxGasLimit field represents the total gas that can be provided in block.                              |

Author: done

> | accumulatedFees       | The accumulatedFees field represents the accumulated fee that was payed in the block.                                                                              |

Reviewer 2: ```suggestion
| accumulatedFees       | The accumulatedFees field represents the accumulated fees that were payed in the block.                                                                              |
```

Author: done

| scheduledData         | The scheduledData contains data about the scheduled execution.                                             |

> | epochStartShardsData  | The epochStartShardsData contains data about the epoch start shards data.                                  |

Author: done

> | developerFees         | The developerFees field represents the fee that was accumulated in the block.                                                                                      |

Reviewer 2: ```suggestion
| developerFees         | The developerFees field represents the fees that were accumulated in the block.                                                                                      |
```

Author: done

> | epochStartShardsData  | The epochStartShardsData is an array of structures that contain epoch-start data for each shard, such as pending miniblocks.                                       |

Reviewer 2: ```suggestion
| epochStartShardsData  | The epochStartShardsData is an array of structures that contains epoch-start data for each shard, such as pending miniblocks.                                       |
```

Author: done

| totalToDistribute                | The totalToDistribute field represents the amount of EGLD that will be distributed to validators/delegators.         |
| totalNewlyMinted                 | The totalNewlyMinted field represents the amount of the new EGLG minted.                                             |
| rewardsPerBlock                  | The rewardsPerBlock field represents the amount of rewards in EGLD per block.                                        |
| rewardsForProtocolSustainability | The rewardsForProtocolSustainability field represents the amount of rewards for the protocol sustainability address. |
| nodePrice                        | The nodePrice field represents the price in EGLD for a validator.                                                    |
| prevEpochStartRound              | The prevEpochStartRound field represents the round of the previous epoch start block.                                |
| prevEpochStartHash               | The prevEpochStartHash field represents the hash of the previous epoch start block.                                  |

## Query examples

> #### Fetch blocks for a shard 

Reviewer: only `#` needed

Author: `###`

> The `metachain` block (`shardId:4294967295`) with field `epochStartBlock:true` has the field `epochStartInfo` populated with the next data:

Reviewer 2: ```suggestion
A `metachain` block (`shardId:4294967295`) with field `epochStartBlock:true` will have the `epochStartInfo` field populated with:
```

Author: done

In order to fetch the latest blocks from one shard, one has to do a query that matches the field `shardId`.
```
curl --request GET \
  --url ${ES_URL}/blocks/_search \
  --header 'Content-Type: application/json' \
  --data '{
    "query": {
        "match": {

>             "shardId": "1"

Reviewer: so even if the value is a number, it should be stringified?

Author: it can be a number or string it works anyway.

> | totalNewlyMinted                 | The totalNewlyMinted field represents the amount of the new EGLG minted.                                             |

Reviewer 2: ```suggestion
| totalNewlyMinted                 | The totalNewlyMinted field represents the amount of the newly minted EGLD.                                             |
```

Author: done

> | nodePrice                        | The nodePrice field represents the price in EGLD for a validator.                                                    |

Reviewer 2: ```suggestion
| nodePrice                        | The nodePrice field represents the EGLD amount required to run a validator.                                                    |
```

> In order to fetch the latest blocks from one shard, one has to do a query that matches the field `shardId`.

Reviewer 2: ```suggestion
In order to fetch the latest blocks from a shard, one has to do a query that matches the field `shardId`.
```

Author: done

| proposer              | The proposer field represents the index of the validator that proposed this block.                                                                                 |
| validators            | The validators field is an array that contains the indices of the validators that signed the block. The indices are to be found in the validators index.           |
| pubKeyBitmap          | The pubKeyBitmap field represents the pub key bitmap.                                                                                                              |
| size                  | The size field represents the size of the block in bytes.                                                                                                          |
| sizeTxs               | The sizeTxs field holds the size of the block's transactions in bytes.                                                                                             |
| timestamp             | The timestamp field represents the timestamp when the block was proposed and executed.                                                                             |

> | stateRootHash         | The stateRootHash field represents the trie' state root hash when the block was proposed and executed.                                                             |

Author: done
